spyder/plugins/completion/kite/plugin.py

```diff
@@ -34,16 +34,32 @@ def __init__(self, parent):
         enable_code_snippets = CONF.get('lsp-server', 'code_snippets')
         self.client = KiteClient(None, enable_code_snippets)
         self.kite_process = None
+        self.main_window_visible = False
+        self.kite_initialized = False
+        self.onboarding_shown = False
         self.client.sig_client_started.connect(self.http_client_ready)
         self.client.sig_response_ready.connect(
             functools.partial(self.sig_response_ready.emit,
                               self.COMPLETION_CLIENT_NAME))
+        self.main.sig_setup_finished.connect(self.mainwindow_setup_finished)
```

Comment:
Maybe we could connect the new `sig_setup_finished` signal directly with the `_kite_onboarding` method (and remove then the `main_window_visible` attribute and the `main_window_setup_finished` method)

Reply:
My thinking is that `self.client.start()` is starting the client async and will trigger `http_client_ready` at some point. `sig_setup_finished` is triggered independently from this. So both may happen in any order and this shouldn't trigger onboarding just once (not 0 or 2 times).
This might be too cautious if there's some synchronization happening which I didn't notice.
What do you think?

Reply:
Ok, makes sences, thanks for the explanation :+1: